TAO/orbsvcs/tests/ImplRepo/run_test.pl

```diff
@@ -638,6 +638,371 @@ sub nestea_test
+    }
+    my $act_imriorfile = $act->LocalFile ($imriorfile);
+    my $ti_imriorfile = $ti->LocalFile ($imriorfile);
+    my $act_actiorfile = $act->LocalFile ($actiorfile);
+
+    my @a_srv_imriorfile;
+    my @imr_airplaneiorfile;
+    my @a_srv_airplaneiorfile;
+    for ($index = 0; $index < $num_srvr; ++$index) {
+        push(@a_srv_imriorfile, $a_srv[$index]->LocalFile ($imriorfile));
+        push(@imr_airplaneiorfile, $imr->LocalFile ($airplaneiorfile[$index]));
+        push(@a_srv_airplaneiorfile, $a_srv[$index]->LocalFile ($airplaneiorfile[$index]));
+        $a_srv[$index]->DeleteFile ($imriorfile);
+        $a_srv[$index]->DeleteFile ($airplaneiorfile[$index]);
+        $a_cli[$index]->DeleteFile ($airplaneiorfile[$index]);
+        $imr->DeleteFile ($airplaneiorfile[$index]);
+    }
+    $imr->DeleteFile ($imriorfile);
+    if ($backing_store_flag eq "--directory") {
+        cleanup_replication($imr_storefile);
+    }
+    $act->DeleteFile ($imriorfile);
+    $ti->DeleteFile ($imriorfile);
+    $act->DeleteFile ($actiorfile);
+
+    my $bin_imr_host = $bin_imr->HostName ();
+    my $imr_initref = "-orbinitref ImplRepoService=corbaloc::$bin_imr_host:8888/ImplRepoService";# -ORBDebugLevel 10";
+
+    ## Be sure to start the ImR on a consistent endpoint, so that any created IORs
+    ## remain valid even if the ImR restarts.
+    my $imr_arguments = "-orbendpoint iiop://:8888 $backing_store_flag $imr_storefile -d $test_debug_level -o $imr_imriorfile $imr_refstyle";
+    $IMR->Arguments ("$imr_arguments -e ");
+    test_info("starting IMR=" . $IMR->CommandLine() . "\n");
+    my $IMR_status = $IMR->Spawn ();
+    if ($IMR_status != 0) {
+        print STDERR "ERROR: ImR Service returned $IMR_status\n";
+        return 1;
+    }
+    if ($imr->WaitForFileTimed ($imriorfile,
+                                $imr->ProcessStartWaitInterval()) == -1) {
+        print STDERR "ERROR: cannot find file <$imr_imriorfile>\n";
+        $IMR->Kill (); $IMR->TimedWait (1);
+        return 1;
+    }
+    if ($imr->GetFile ($imriorfile) == -1) {
+        print STDERR "ERROR: cannot retrieve file <$imr_imriorfile>\n";
+        $IMR->Kill (); $IMR->TimedWait (1);
+        return 1;
+    }
+    if ($act->PutFile ($imriorfile) == -1) {
+        print STDERR "ERROR: cannot set file <$act_imriorfile>\n";
+        $IMR->Kill (); $IMR->TimedWait (1);
+        return 1;
+    }
+    if ($ti->PutFile ($imriorfile) == -1) {
+        print STDERR "ERROR: cannot set file <$ti_imriorfile>\n";
+        $IMR->Kill (); $IMR->TimedWait (1);
+        return 1;
+    }
+
+    for ($index = 0; $index < $num_srvr; ++$index) {
+        if ($a_srv[$index]->PutFile ($imriorfile) == -1) {
+            print STDERR "ERROR: cannot set file <$a_srv_imriorfile[$index]>\n";
+            $IMR->Kill (); $IMR->TimedWait (1);
+            return 1;
+        }
+    }
+
+    $ACT->Arguments ("-d $test_debug_level -o $act_actiorfile -ORBInitRef ImplRepoService=file://$act_imriorfile");
```

Comment:
I added some logic to also kill and restart the activator, the server is killed when the activator is down